### PR TITLE
fix editbox can't input on macos 10.12

### PR DESCRIPTION
**Pull request by zilongshanren**
Fix issue: https://github.com/cocos2d/cocos2d-x/issues/16779

### CODE REVIEW

File: cocos/ui/UIEditBox/Mac/CCUIEditBoxMac.mm [hunk 1]
```diff
@@ -135,6 +135,8 @@ - (NSWindow *)window
 - (void)openKeyboard
 {
     [self.window.contentView addSubview:self.textInput];
+    //FIXME: it will cause a few OpenGL error once, should be fix when glfw version upgraded.
```
Comment (dumganhar) [inline cocos/ui/UIEditBox/Mac/CCUIEditBoxMac.mm:138]: errors? fixed? is upgraded?

Comment (zilongshanren) [reply cocos/ui/UIEditBox/Mac/CCUIEditBoxMac.mm:138]: There is a few OpenGL errors, but it doesn't affect the rendering and these error only occurs once. I think the problem should be resolved in the glfw upstream. But I can't handle it properly at the moment. This is a workaround, but at least it works well except for the error message.

Comment (zilongshanren) [reply cocos/ui/UIEditBox/Mac/CCUIEditBoxMac.mm:138]: @dumganhar 
In the future, when glfw fixed the EditBox input issue on mac os, we should remove this hack code.

Comment (zilongshanren) [reply cocos/ui/UIEditBox/Mac/CCUIEditBoxMac.mm:138]: @dumganhar 
The glfw patch is here https://github.com/cocos2d/cocos2d-x-3rd-party-libs-src/blob/v3/contrib/src/glfw/fix-mac-os-10-12.patch

Comment (minggo) [reply cocos/ui/UIEditBox/Mac/CCUIEditBoxMac.mm:138]: I think you can send a patch to GLFW to discuss with them.

Comment (zilongshanren) [reply cocos/ui/UIEditBox/Mac/CCUIEditBoxMac.mm:138]: @minggo  
Already opened a ticket in  https://github.com/glfw/glfw/issues/892  But it seems not get active response.

Comment (minggo) [reply cocos/ui/UIEditBox/Mac/CCUIEditBoxMac.mm:138]: Looks good to me. I will merge it, but i think it is better to create an issue in cocos2d-x that revert the codes when GLFW fixes the issue.

Comment (rockingdice) [reply cocos/ui/UIEditBox/Mac/CCUIEditBoxMac.mm:138]: can you share a working GLFW library file?

Comment (rockingdice) [reply cocos/ui/UIEditBox/Mac/CCUIEditBoxMac.mm:138]: well I compiled the GLFW library by myself but, it is still not working yet. I also try the combination of cocos2d-x side  changes and glfw changes.

Comment (zilongshanren) [reply cocos/ui/UIEditBox/Mac/CCUIEditBoxMac.mm:138]: @rockingdice
You could got a GLFW library from here: https://github.com/cocos2d/cocos2d-x-3rd-party-libs-bin

**Pull request merged**